crates/runestick/src/modules/result.rs

```diff
@@ -38,18 +40,33 @@ fn expect_impl(result: Result<Value, Value>, message: &str) -> Result<Value, VmE
 }
 
 fn and_then_impl(
-    this: &Result<Value, Value>,
+    this: Result<Value, Value>,
     then: Function,
 ) -> Result<Result<Value, Value>, VmError> {
     match this {
         Ok(v) => Ok(then.call::<_, _>((v,))?),
-        Err(e) => Ok(Err(e.clone())),
+        Err(e) => Ok(Err(e)),
     }
 }
 
-fn map_impl(this: &Result<Value, Value>, then: Function) -> Result<Result<Value, Value>, VmError> {
+fn map_impl(this: Result<Value, Value>, then: Function) -> Result<Result<Value, Value>, VmError> {
     match this {
         Ok(v) => Ok(Ok(then.call::<_, _>((v,))?)),
-        Err(e) => Ok(Err(e.clone())),
+        Err(e) => Ok(Err(e)),
     }
 }
+
+fn as_ref_impl(result: &mut Result<Value, Value>) -> Result<Result<Value, Value>, VmError> {
```

Comment:
I believe the effective outcome of this function is the same as cloning `result` and its variants twice.

1. `&result` auto-derefs to a reference of result and we pattern match over its variants.
2. The implementation of `ToValue` for `&Value` [is to clone it](https://github.com/rune-rs/rune/blob/main/crates/runestick/src/to_value.rs#L43).
3. We clone the value in the variant when building a new variant.
4. We then clone the result again at the bottom of the fn?

So I think this effectively has the same effect if we replace it with `Ok(result.clone())` minus a bit of extra cloning unless I'm missing something?